### DISCUSSION TITLE
Blazor server performance issues

### DISCUSSION THREAD

**Discussion by webebox** [dotnet/aspnetcore#29418] [General]
Its a Blazor server project with about 100 .razor pages, When a change occurs on one of the pages, it takes about 19 seconds to rebuild pages, restart IIS Express and refresh the page. It's awful, I make only a minor change in html, but it takes long time to show the results. The test result is like this :

With 100 razor pages and all css and js references
build : 10 sec
refresh page : 19 sec (include 10 secs for build)

When we remove 90 pages of 100 pages (10 pages remains) :
build : 3 sec
refresh page : 12 sec (include 3 secs for build)

When we remove all css and js references :
build : 3 sec
refresh page : 6 sec (include 3 secs for build)

![image](https://user-images.githubusercontent.com/1694488/105021807-39517f00-5a5e-11eb-823b-95855272b00a.png)

It is not good at all, because the project is growing and finally we will have about 400 pages! and extra css and js references will added. In this case, the time will be much longer for developing.
what's the solution? Thanks

**Comment by davidfowl** [comment]
cc @pranavkm @SteveSandersonMS @DamianEdwards
